### PR TITLE
refactor, chore(egen): refactors code as per new template guidelines, adds code to delete custom job and locally generated files

### PR DESCRIPTION
**Pull request by sumanvita-springml**
This PR includes the following updates:

1. Changes REGION to LOCATION.
2. Removes import os and os.getenv('IS_TESTING') in cleanup section.
3. Adds code to delete locally generated files and custom job in the cleanup section.
4. Re-orders Github and Workbench logos as per the notebook template.
5. Makes structural changes according to new notebook template.

**REQUIRED:** Fill out the below checklists or remove if irrelevant
1. If you are opening a PR for `Official Notebooks` under the [notebooks/official](https://github.com/GoogleCloudPlatform/vertex-ai-samples/tree/main/notebooks/official) folder, follow this mandatory checklist:
- [x] Use the [notebook template](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/notebook_template.ipynb) as a starting point.
- [x] Follow the style and grammar rules outlined in the above notebook template.
- [x] Verify the notebook runs successfully in Colab since the automated tests cannot guarantee this even when it passes.
- [x] Passes all the required automated checks. You can locally test for formatting and linting with these [instructions](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/CONTRIBUTING.md#code-quality-checks).
- [ ] You have consulted with a tech writer to see if tech writer review is necessary. If so, the notebook has been reviewed by a tech writer, and they have approved it.
- [ ] This notebook has been added to the [CODEOWNERS](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/official/CODEOWNERS) file under the `Official Notebooks` section, pointing to the author or the author's team.
- [x] The Jupyter notebook cleans up any artifacts it has created (datasets, ML models, endpoints, etc) so as not to eat up unnecessary resources.

### CODE REVIEW

File: notebooks/official/tensorboard/tensorboard_profiler_custom_training_with_prebuilt_container.ipynb [hunk 1]
```diff
@@ -8,7 +8,7 @@
       },
       "outputs": [],
       "source": [
-        "# Copyright 2022 Google LLC\n",
+        "# Copyright 2024 Google LLC\n",
```
Comment (katiemn) [inline notebooks/official/tensorboard/tensorboard_profiler_custom_training_with_prebuilt_container.ipynb:11]: The year can remain 2022

File: notebooks/official/tensorboard/tensorboard_profiler_custom_training_with_prebuilt_container.ipynb [hunk 2]
```diff
@@ -824,15 +752,14 @@
...
-      "attachments": {},
       "cell_type": "markdown",
       "metadata": {
         "id": "51hKGTbU32Eg"
       },
       "source": [
         "#### Run the custom training job\n",
         "\n",
-        "Next, you run the custom job to start the training job by invoking the method `run`.\n",
+        "Next, you run the custom job to start the training job by invoking the method `run()`.\n",
```
Comment (katiemn) [inline notebooks/official/tensorboard/tensorboard_profiler_custom_training_with_prebuilt_container.ipynb:762]: Can remove 'you' here

File: notebooks/official/tensorboard/tensorboard_profiler_custom_training_with_prebuilt_container.ipynb [hunk 3]
```diff
@@ -45,22 +44,21 @@
         "    </a>\n",
         "  </td>\n",
         "  <td style=\"text-align: center\">\n",
+        "    <a href=\"https://console.cloud.google.com/vertex-ai/workbench/deploy-notebook?download_url=https://raw.githubusercontent.com/GoogleCloudPlatform/vertex-ai-samples/main/notebooks/official/tensorboard/tensorboard_profiler_custom_training_with_prebuilt_container.ipynb\">\n",
+        "      <img src=\"https://lh3.googleusercontent.com/UiNooY4LUgW_oTvpsNhPpQzsstV5W8F7rYgxgGBD85cWJoLmrOzhVs_ksK_vgx40SHs7jCqkTkCk=e14-rj-sc0xffffff-h130-w32\" alt=\"Vertex AI logo\"> <br>\n",
+        "      Open in Vertex AI Workbench\n",
+        "    </a>\n",
+        "  </td> \n",
+        "  <td style=\"text-align: center\">\n",
         "    <a href=\"https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/official/tensorboard/tensorboard_profiler_custom_training_with_prebuilt_container.ipynb\">\n",
```
Comment (kittyabs) [inline notebooks/official/tensorboard/tensorboard_profiler_custom_training_with_prebuilt_container.ipynb:53]: Where is the code for "[Open in Colab Enterprise]?? This should be added

File: notebooks/official/tensorboard/tensorboard_profiler_custom_training_with_prebuilt_container.ipynb [hunk 4]
```diff
@@ -130,185 +125,132 @@
...
-      "attachments": {},
       "cell_type": "markdown",
       "metadata": {
         "id": "3ivZkPUjpaFz"
       },
       "source": [
-        "**4. Setup service account and permissions**\n",
+        "**Setup service account and permissions**\n",
         "\n",
         "A service account will be used to create custom training jobs. If you do not want to use your project's Compute Engine service account, set SERVICE_ACCOUNT to another service account ID. You can create a service account by following the [instructions](https://cloud.google.com/iam/docs/creating-managing-service-accounts#creating)."
```
Comment (kittyabs) [inline notebooks/official/tensorboard/tensorboard_profiler_custom_training_with_prebuilt_container.ipynb:255]: change "will be" to "is":

- A service account is used to create custom training jobs

**Review by kittyabs — COMMENTED** [review]
Add "Open in Colab Enterprise" button, and other edits

File: notebooks/official/tensorboard/tensorboard_profiler_custom_training_with_prebuilt_container.ipynb (hunk 3 above)
Comment (sumanvita-springml) [reply notebooks/official/tensorboard/tensorboard_profiler_custom_training_with_prebuilt_container.ipynb:51]: code for "[Open in Colab Enterprise] was already present. However, I missed changing the names for each logo as per notebook template guidelines. Made the edits accordingly and pushed the changes

**Review by katiemn — APPROVED** [review]

**Pull request merged**